frontend/src/features/admin-form/responses/ResponsesPage/storage/useSecretKey.ts

```diff
@@ -0,0 +1,75 @@
+import { useEffect, useState } from 'react'
+import { BroadcastChannel } from 'broadcast-channel'
+
+import { useHasChanged } from '~hooks/useHasChanged'
+
+const SECRETKEY_BROADCAST_KEY = 'formsg_private_key_sharing'
+
+type SecretKeyBroadcastMessage =
+  | {
+      formId: string
+      action: 'requestKey'
+    }
+  | {
+      formId: string
+      action: 'broadcastKey'
+      secretKey: string
+    }
+
+// BroadcastChannel will only broadcast the message to scripts from the same origin
+// (i.e. https://form.gov.sg in practice) so all data should be controlled by scripts
+// originating from FormSG. This does not store any data in browser-based storage
+// (e.g. cookies or localStorage) so secrets would not be retained past the user closing
+// all Form tabs containing the form.
+const secretKeyChannel = new BroadcastChannel<SecretKeyBroadcastMessage>(
+  SECRETKEY_BROADCAST_KEY,
+)
+
+export const useSecretKey = (formId: string) => {
```

Comment:
this is really just for broadcast right? maybe `useSecretKeyBroadcast` to make the purpose clear

Reply:
also to generate and keep state of secret key, so i think i'll keep the name